libs/wire-api/src/Wire/API/User/RichInfo.hs

```diff
@@ -64,17 +64,48 @@ import Wire.API.Arbitrary (Arbitrary (arbitrary))
 --------------------------------------------------------------------------------
 -- RichInfo
 
--- | Represents all the ways we can recieve 'RichInfo' from a SCIM peer.  'richInfoMap'
--- represents fields given under 'richInfoMapURN' as a JSON object.  'richInfoAssocList'
--- represents fields given under 'richInfoAssocListURN' as an assoc list.  Use the much
--- simpler 'RichInfoAssocList' if you can.  Use 'toRichInfoAssocList' to turn 'RichInfo' into
--- 'RichInfoAssocList'.
-data RichInfo = RichInfo
+-- | A 'RichInfoAssocList' that parses and renders as 'RichInfoMapAndList'.
+newtype RichInfo = RichInfo {unRichInfo :: RichInfoAssocList}
+  deriving stock (Eq, Show, Generic)
+
+instance ToJSON RichInfo where
+  toJSON = toJSON . fromRichInfoAssocList . unRichInfo
+
+instance FromJSON RichInfo where
+  parseJSON = fmap (RichInfo . toRichInfoAssocList) . parseJSON
+
+instance Arbitrary RichInfo where
+  arbitrary = RichInfo <$> arbitrary
+
+instance Monoid RichInfo where
+  mempty = RichInfo mempty
+
+instance Semigroup RichInfo where
+  RichInfo a <> RichInfo b = RichInfo $ a <> b
+
+--------------------------------------------------------------------------------
+-- RichInfoMapAndList
+
+-- | Represents all the ways we can recieve 'RichInfo' from a SCIM peer.
+--
+-- 'richInfoMap' represents fields given under 'richInfoMapURN' as a JSON object (use case:
+-- enterprise extensions as sent by eg. microsoft azure).
+--
+-- 'richInfoAssocList' represents fields given under 'richInfoAssocListURN' as an assoc list
+-- (use case: wire native code; we used this so we can give the client arbitrary order in
+-- which to show the rich info).
+--
+-- Use the much simpler 'RichInfoAssocList' if you can.  Use 'toRichInfoAssocList' to turn
```

Comment:
I think you mean `RichInfo` here. `RIchInfoAssocList` doesn't exist anymore per this commit right

Reply:
oh wait; `RichInfo` is a newtype wrapper around `RichInfoAssocList`. This is all a bit confusing :/

Reply:
5dc68e9877fb2be231b73b5e17dc678ee04fce88  :)  sorry